File: day01/main.py
filename = "./input"
elves: list[list[int]] = []

with open(filename, "r") as file:
    current_elf = []
    for line in file.readlines():
        if line == "\n":
            elves.append(current_elf)
            current_elf = []
            continue
        current_elf.append(int(line.strip()))


max = 0
for elf in elves:
    calories = sum(elf)
    if calories > max:
        max = calories

print(f"Puzzle 1: {max}")

sums = []
for elf in elves:
    sums.append(sum(elf))

sums.sort(reverse=True)

print(f"Puzzle 2: {sum(sums[:3])}")

File: day01/main.go
package main

import (
	aoc "aoc/utils"
	"fmt"
	"sort"
	"strconv"
)

const filepath = "./input"

func main() {
	lines, err := aoc.ReadLines(filepath)
	aoc.Handle(err)

	elves := [][]int{}

	calories := []int{}
	for _, line := range lines {
		if line == "" {
			elves = append(elves, calories)
			calories = []int{}
		} else {
			number, err := strconv.Atoi(line)
			aoc.Handle(err)
			calories = append(calories, number)
		}
	}

	var maxCalories int
	for _, elf := range elves {
		calories := aoc.Sum(elf)
		if calories > maxCalories {
			maxCalories = calories
		}
	}
	fmt.Println("Puzzle 1:", maxCalories)

	elveCalories := []int{}
	for _, elf := range elves {
		elveCalories = append(elveCalories, aoc.Sum(elf))
	}
	sort.Ints(elveCalories)

	fmt.Println("Puzzle 2:", aoc.Sum(elveCalories[len(elveCalories)-3:]))
}
